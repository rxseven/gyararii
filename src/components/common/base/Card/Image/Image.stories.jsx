import React from 'react';
import { storiesOf } from '@storybook/react';

import Image from './index';

const title = 'Components/Common/Base/Card/Image';

storiesOf(title, module).add('default', () => (
  <Image src="https://via.placeholder.com/480x320?text=Image-001" />
));
