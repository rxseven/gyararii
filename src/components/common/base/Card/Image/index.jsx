import PropTypes from 'prop-types';
import React from 'react';
import Img from 'react-image';

const propTypes = {
  alt: PropTypes.string,
  src: PropTypes.string.isRequired
};

const defaultProps = {
  alt: ''
};

function Image({ className, ...props }) {
  return <Img {...props} className={`card-img ${className}`} />;
}

Image.propTypes = propTypes;
Image.defaultProps = defaultProps;

export default Image;
