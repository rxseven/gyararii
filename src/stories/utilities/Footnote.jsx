import styled from 'styled-components';

const Footnote = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

export default Footnote;
