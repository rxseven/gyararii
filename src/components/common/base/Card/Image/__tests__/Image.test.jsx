import Img from 'react-image';

import { factory } from 'tests/utilities';
import Image from '../index';

// Mock
jest.mock('react-image');

// Arrange
const context = expect.any(Object);
const seed = {
  alt: '',
  className: 'card-img',
  src: '/gallery/image-001.png'
};
const source = { src: seed.src };
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Image, source, props);
}

// Test suites
describe('<Card.Image />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed image', () => {
    const expected = { called: { ...input } };
    setup();

    expect(Img).toHaveBeenCalledWith(expected.called, context);
  });

  it('should render with custom "alt" prop', () => {
    const props = { alt: 'Gallery' };
    const expected = { called: { ...input, ...props } };
    setup(props);

    expect(Img).toHaveBeenCalledWith(expected.called, context);
  });

  it('should render with custom class names', () => {
    const props = { className: 'custom-class' };
    const expected = {
      called: {
        ...input,
        className: `card-img ${props.className}`
      }
    };
    setup(props);

    expect(Img).toHaveBeenCalledWith(expected.called, context);
  });
});
