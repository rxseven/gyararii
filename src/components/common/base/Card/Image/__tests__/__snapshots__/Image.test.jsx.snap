// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card.Image /> Snapshot tests should render correctly 1`] = `null`;
